README.md

```diff
@@ -8,39 +8,352 @@ Macro](https://img.shields.io/badge/babel--macro-%F0%9F%8E%A3-f5da55.svg?style=f
+
+1- **Server-side Computation with Subscription:**
+```javascript
+// formattedTime.ssr-computation.js
 
-↓ ↓ ↓ ↓ ↓ ↓
+import { DateTime } from "luxon";
 
-```ruby
-import { useSSRComputationClient } from "use-ssr-computation/lib/useSSRComputation";
-const x = useSSRComputationClient("./a.ssr-computation");
+export const compute = () => {
+  return DateTime.now().toLocaleString(DateTime.TIME_SIMPLE);
+};
+
+let timerId;
+
+export const subscribe = (getCurrentResult, next) => {
+  if (timerId) clearInterval(timerId);
+
+  timerId = setInterval(() => {
+    const newValue = compute();
+    if (newValue !== getCurrentResult()) {
+      next(newValue);
+    }
+  }, 60000);  // Update every minute
+
+  return {
+    unsubscribe: () => {
+      clearInterval(timerId);
+    },
+  };
+};
 ```
+2-**Application Integration:**
+```javascript
+// App.js
+
+import React, { useEffect } from "react";
+import { useSSRComputation, fetchSubscriptions } from "use-ssr-computation.macro";
+
+const App = () => {
+  const formattedTime = useSSRComputation("./formattedTime.ssr-computation");
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      fetchSubscriptions();
+    }, 60000); // Subscribe after 1 minute
+
+    return () => clearTimeout(timeoutId);
+  }, []);
+
+  return (
+    <div>
+      Current time is: {formattedTime}
+    </div>
+  );
+};
+
+export default App;
+```
+
+The computation module and the `luxon` package are not loaded immediately, thereby optimizing initial page load times. Only after 1 minute (when the `fetchSubscriptions` function is called) will the library and module be dynamically loaded and the subscription set up to update the time.
+
+With this mechanism, we've effectively lazy-loaded our time formatting operation, deferring the load of `luxon` and ensuring minimal performance impact during the critical initial page render.
+
+### Performance Gains
+Why go through this trouble?
+
+- Lazy Loading: Import heavy libraries only when they're truly necessary.
+- Reduced Network Payload: Only essential data is sent over the wire.
+- Optimized CPU Utilization: Client devices do less computational work, leading to better responsiveness.
+
+## Installation
+### Install Packages
+**NPM**
+```bash
+npm install babel-plugin-macros @shakacode/use-ssr-computation.macro @shakacode/use-ssr-computation.runtime
+```
+**Yarn**
+```bash
+yarn add babel-plugin-macros @shakacode/use-ssr-computation.macro @shakacode/use-ssr-computation.runtime
+```
+
+### Configure Babel
+1- Add the following to your `.babelrc` that's responsible for compiling your server-bundle file:
+
+```json
+{
+  "plugins": [
+    "macros",
+    {
+      "useSSRComputation": {
+        "side": "server"
+      }
+    }
+  ]
+}
+```
+
+2- Add the following to your `.babelrc` that's responsible for compiling your client-bundle file:
+
+```json
+{
+  "plugins": [
+    "macros",
+    {
+      "useSSRComputation": {
+        "side": "client"
+      }
+    }
+  ]
+}
+```
```

Comment:
It depends on the webpack configuration used in the app. I already left a reference to the `RORP` dummy app. So anyone can see how it can be implemented.